woocommerce-services.php

```diff
@@ -1440,6 +1441,25 @@ private function get_i18n_json() {
 			return '{}';
 		}
 
+		public function edit_orders_page_actions() {
+			if ( is_admin() ) {
+				$screen = get_current_screen();
+				if ( ! $screen || ! isset( $screen->id ) ) {
+					return false;
+				}
+
+				if ( 'edit-shop_order' !== $screen->id ) {
```

Comment:
It doesn't work for me, because the $screen->id in the orders page, in my setup is called `woocommerce_page_wc-orders`, I'm not sure why is different? I'm using WooCommerce `v8.9.3`, not sure if that changes a lot in `v9.0.0`

Reply:
🤔 Mine is also WC v8.9.3. 

It looks like this is related to HPOS https://github.com/Automattic/woocommerce-services/pull/2692/files. I will borrow the logic there.

Reply:
Yup I can confirm HPOS returns "woocommerce_page_wc-orders" instead. Fixed in efc2229f. Tested against 2 installations, one with HPOS, and the other without.

Reply:
Nope, that did not fix it completely. The modal now opens inside a single order as well.  I tried updating it to
```diff
if (OrderUtil::custom_orders_table_usage_is_enabled()) {
+                                       // Orders list page when using HPOS.
+                                       if (wc_get_page_screen_id( 'shop_order' ) === $screen->id) {
+                                               $should_display_wcst_to_wcshipping_migration_notice = true;
+                                       }
+                               } else {
+                                       // Orders list page when not using HPOS.
+                                       if ('shop_order' === $screen->post_type && in_array( $screen->base, array( 'edit' ), true )) {
+                                               $should_display_wcst_to_wcshipping_migration_notice = true;
+                                       }
+                               }
+
+                               if ($should_display_wcst_to_wcshipping_migration_notice) {
```
But that doesn't fix it 100% either. This also makes it very complicated.

Reply:
Ok, I added a `is_on_order_list_page()` function to check if we are on the order list page. The admin notice should show only when it's on the order listing page (HPOS or not).

Fixed in 0e270e03.